src/analysis/output.js

```diff
@@ -0,0 +1,104 @@
+// @flow
+
+/**
+ * This module defines a rich output format for cred scores, so that we can use
+ * it to drive UIs and data analysis.
+ */
+import type {Alias} from "../plugins/identity/alias";
+import type {EdgeWeight} from "../core/weights";
+import type {PluginDeclaration} from "./pluginDeclaration";
+
+// unenforced
+export type Integer = number;
+export type TimestampMs = Integer;
+export type Index = Integer;
+
+/**
+ * Describes an individual node in the contribution graph.
+ * Includes the information in the raw Graph node, along with scoring
+ * information, i.e. its total cred, and the amount of cred that it minted.
+ *
+ * While we are in TimelineCred, the "cred" field will have the total cred,
+ * summed across all time slices. Once we migrate to CredRank, it will just be
+ * the cred directly calculated by the CredRank algorithm.
+ */
+export type OutputNode = {|
+  +address: $ReadOnlyArray<string>,
+  +cred: number,
+  +mint: number,
+  +timestamp: TimestampMs | null,
+  +description: string,
+|};
+
+/**
+ * The first version of the output format will have the nodes, and the plugin
+ * declarations. Including the declarations makes it easy for consumers to do
+ * analyses like: how much cred was minted within each plugin? Within each
+ * contribution type? Etc.
+ */
+export type OutputV1 = {|
+  // Ordered by address
+  +orderdNodes: $ReadOnlyArray<OutputNode>,
+  +plugins: $ReadOnlyArray<PluginDeclaration>,
+|};
+
+/**
+ * Extra data for Contributors. Note that each Contributor corresponds
+ * to a specific node in the graph, which is retrievable via the NodeIndex.
+ *
+ * For contributors, we'll give the cred-over-time. We don't include this
+ * information for regular contribution nodes, both because it's not that
+ * interesting, and because once we migrate to CredRank, we won't have that
+ * level of resolution anymore.
+ *
+ * While we are in TimelineCred, the credByTime will be the raw cred that the
+ * contributor recieved in each time slice. Once we switch to TimelineCred, the
+ * credByTime will instead be the amount of cred that flowed from the
+ * contributor's epoch node for each time period.
```

Comment:
Is it perhaps clearer to phrase this as “the amount of cred at the
contributor’s epoch node for each time period” rather than “…that flowed
from…” the epoch node? It’s technically equivalent for a stationary
distribution, but imho easier to think about instantaneously.

Reply:
No, it's not equivalent. The contributor's epoch node will flow to other nodes: e.g. thru the webbing edges, and (if sponsorship has occured) to the sponsor. Therefore, the contributor's share of their cred from that epoch node is the amount that flowed to the contributor from that epoch node, not the cred at the epoch node.

Updated the comment to make this clearer.